### PR TITLE
feat!: Allows `ERC725X.executeBatch` path in LSP6 Key Manager

### PR DESCRIPTION
**Pull request by b00ste**
# What does this PR introduce?

## 🚀 Feature

This PR Introduces the verification for the path`ERC725X.executeBatch` in the LSP6 Key Manager.

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-8798 add executeBatch path in Key Manager](https://app.clickup.com/t/2645698/DEV-8798)

File: contracts/LSP6KeyManager/LSP6KeyManagerCore.sol [hunk 1]
```diff
@@ -581,8 +585,42 @@ abstract contract LSP6KeyManagerCore is
...
+            ) {
+                revert ERC725X_ExecuteParametersLengthMismatch();
+            }
+
+            if (operationTypes.length == 0) {
+                revert ERC725X_ExecuteParametersEmptyArray();
+            }
+
+            for (uint256 ii = 0; ii < operationTypes.length; ii++) {
+                _verifyCanExecute(
```
Comment (CJ42) [inline contracts/LSP6KeyManager/LSP6KeyManagerCore.sol:614]: Should we do like this for consistency with the others for now?

```suggestion
                LSP6ExecuteModule._verifyCanExecute(
```

File: tests/LSP20CallVerification/LSP6/Interactions/ERC725XExecuteBatch.test.ts [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+      const values = [0, 0, 0];
+
+      const payloads = [lyxDaiTransferPayload, metaCoinTransferPayload, rLYXTransferPayload];
+
+      await universalProfile
+        .connect(context.mainController)
+        .executeBatch(operationTypes, targets, values, payloads);
+
+      expect(await lyxDaiToken.balanceOf(recipient)).to.equal(
```
Comment (CJ42) [inline tests/LSP20CallVerification/LSP6/Interactions/ERC725XExecuteBatch.test.ts:180]: Let's add tests to check that the token balance of the UP also decreased.

File: tests/LSP20CallVerification/LSP6/Interactions/ERC725XExecuteBatch.test.ts [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+        );
+
+      // CHECK that token contract has been deployed
+      await expect(tx)
+        .to.emit(context.universalProfile, 'ContractCreated')
+        .withArgs(
+          OPERATION_TYPES.CREATE,
+          ethers.utils.getAddress(futureTokenAddress),
+          0,
+          ethers.utils.hexZeroPad('0x00', 32),
```
Comment (CJ42) [inline tests/LSP20CallVerification/LSP6/Interactions/ERC725XExecuteBatch.test.ts:243]: What is this argument again? Salt I think?

```suggestion
          ethers.utils.hexZeroPad('0x00', 32), // salt
```

File: contracts/LSP6KeyManager/LSP6KeyManagerCore.sol [hunk 4]
```diff
@@ -581,8 +585,42 @@ abstract contract LSP6KeyManagerCore is
...
+                revert ERC725X_ExecuteParametersLengthMismatch();
+            }
+
+            if (operationTypes.length == 0) {
+                revert ERC725X_ExecuteParametersEmptyArray();
+            }
+
+            for (uint256 ii = 0; ii < operationTypes.length; ii++) {
+                LSP6ExecuteModule._verifyCanExecute(
+                    _target,
```
Comment (YamenMerhi) [inline contracts/LSP6KeyManager/LSP6KeyManagerCore.sol:615]: instead of querying the storage, we can use the param `targetContract` ?

**Review by YamenMerhi — APPROVED** [review]

**Review by skimaharvey — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**